app/Library/Scripts/Tasks/RunScripts.cs

```diff
@@ -42,5 +42,12 @@ public override void Execute()
                 }
             }
         }
+
+        private string GetPythonArguments(string fileName)
+        {
+            if (Config.PythonMode == PythonMode.Script) return fileName;
+            var extension = Path.GetExtension(fileName);
+            return String.Format("-m {0}", fileName.Replace("\\", ".").Substring(0, fileName.Length - extension.Length));
+        }
```

Comment:
I'm pretty sure this function will return the wrong filename on non-Windows because of the `.Replace("\\", ".")`. I think the best way to fix this is to factor GetPythonArguments to be a Task (sub-task of RunScripts) and add some tests to make sure that it returns the correct arguments string given a python mode and filename. The logic will need to be changed to use the `System.IO.Path` class to deal with the filename in a cross-platform way.

This is really the only missing set of tests. I'd like to have some full integration tests that use actual files but it's really not necessary to test that python works. We just need to make sure `please` knows how to call python correctly which `GetPythonArgumentsTest` would do.